packages/ember-testing/lib/test.js

```diff
@@ -292,35 +292,26 @@ function helper(app, name) {
       return fn.apply(app, args);
     }
 
-    if (!lastPromise) {
-      // It's the first async helper in current context
-      lastPromise = fn.apply(app, args);
-    } else {
-      // wait for last helper's promise to resolve and then
-      // execute. To be safe, we need to tell the adapter we're going
-      // asynchronous here, because fn may not be invoked before we
-      // return.
-      Test.adapter.asyncStart();
-      run(function() {
-        lastPromise = Test.resolve(lastPromise).then(function() {
-          try {
-            return fn.apply(app, args);
-          } finally {
-            Test.adapter.asyncEnd();
-          }
-        });
+    // wait for last helper's promise to resolve and then
+    // execute. To be safe, we need to tell the adapter we're going
+    // asynchronous here, because fn may not be invoked before we
+    // return.
+    Test.adapter.asyncStart();
+    return run(function() {
```

Comment:
This helper could be called directly from a test (no run loop). I think removing it could trigger an autorun because `Promise.resolve` requires a run loop.